### DISCUSSION TITLE
Filtering http responses in graphql

### DISCUSSION THREAD

**Discussion by Bazalii** [quarkusio/quarkus#29232] [Q&A] [answered]
Is there an option to add filter that will handle http responses using filters? 

For example, I want to log information about response(status code, returned entity, etc.) but I don't get how can I make @RouteFilter to be called after response because priority sets subsequence only for filters that will be called before controller method call and I need this filter to be called after controller method. Usage of @ServerResponseFilter and @ServerRequestFilter wasn't succesfull. Application just didn't see these filters at all...

[a comment by an automated account is not shown]

**Comment by phillip-kruger** [comment, marked as answer]
GraphQL does not use Servlets or JAX-RS, but run directly on Vert.x HTTP. So whatever you can do with Vert.x should work. You can also use CDI Interceptor, but that will be per operation, and not the full execution.

If you just want to see the payload in and out, you can also use the log-payload config option: https://quarkus.io/guides/smallrye-graphql#quarkus-smallrye-graphql_quarkus.smallrye-graphql.log-payload

**Reply by Bazalii** [reply]
@phillip-kruger but actually is there an option to handle a response body using RoutingFilters in Quarkus? I found that there is BodyEndHandler in Vert.X HTTP that can be easily called after the whole body is written to the wire but I do not understand how to get the response body because it is not stored in the routing context

**Reply by phillip-kruger** [reply]
Maybe @cescoffier can help here ?

**Reply by cescoffier** [reply]
I don't believe you will be able to intercept the body as I suspect graphQL is writing into the response (using the `.end` method), which means you cannot manipulate it (the consumer may have already received parts).

**Reply by Bazalii** [reply]
@cescoffier okay. Is it possible to just get the response body using BodyEndHandler to log it? I do not understand if there is such an option because it seems like that response body is just not stored by Vert.X

**Reply by cescoffier** [reply]
in general, you should not store the potentially large body in memory, and it can contain sensible data.
If you look at https://github.com/vert-x3/vertx-web/blob/master/vertx-web/src/main/java/io/vertx/ext/web/handler/impl/LoggerHandlerImpl.java, you will see that everything else can be logged, but not the body. 

If you need that kind of interception, it must be done in the graphQL engine (less generic than HTTP).

**Reply by Bazalii** [reply]
@cescoffier Thank you!
